rioxarray/rioxarray.py

```diff
@@ -918,7 +918,8 @@ def isel_window(self, window):
         row_slice = slice(int(row_start), int(row_stop))
         col_slice = slice(int(col_start), int(col_stop))
         return (
-            self._obj.isel({self.y_dim: row_slice, self.x_dim: col_slice})
+            self._obj.rio.pad_box(*rasterio.windows.bounds(window, self.transform(recalc=True)))
```

Comment:
I would prefer there to be a `pad` kwarg that defaults to False. Then, do something like this:
```python
array_subset = (
    ...
)
if not pad:
   return array_subset
return array_subset.rio.rio.pad_box(
    *rasterio.windows.bounds(window, self.transform(recalc=True))
)

Reply:
okay sounds good, but I think the padding needs to occur prior to the rest of the logic in isel_window, otherwise the xarray isel function will have already filtered out the relevant data by this point in the code

Reply:
@snowman2 let me know what you think about the changes I just pushed, should I also call copy prior to pad_box?

Reply:
> I think the padding needs to occur prior to the rest of the logic in isel_window, otherwise the xarray isel function will have already filtered out the relevant data by this point in the code

Have you verified this is the case?

Reply:
> should I also call copy prior to pad_box?

No, I don't think that is necessary.

Reply:
My main concern about padding first is that  the indexes used by `isel` are based on original array. If you pad the array, the indexes are shifted and your are likely pulling incorrect data.

Reply:
IIRC, the origin of the bug was due to the ranges from the window going further negative than -1 on either of the two axes, and then the isel() call would return a dataarray with 0 range on that dimension, so I don't know if padding after that point would work as only nans are present in the dataarray at that point. I agree with making a test though, that will clarify many aspects of the issue

Reply:
in a debugger I still see the issue that xarray's isel is not able to accept any slices with negative starts, so I think the usage of pad_box is not totally appropriate, or I am going to have to inspect and modify the user's provided window

Reply:
for example try `anyrioxarray.isel({'y': slice(-5,10),'x': slice(-5,10)}).value` it will return an empty array

Reply:
> I still see the issue that xarray's isel is not able to accept any slices with negative starts

What you will likely need to do is where there is the `ceil/floor` sections, set the slice start to 0 or None if the start is less than 0 if having a negative index breaks it.

Reply:
This seems to work for me:
```python
        row_start = 0 if row_start < 0 else math.floor(row_start)
,,,
        col_start = 0 if col_start < 0 else math.floor(col_start)
```

A good test for this is to check that it matches the rasterio behavior:
```python
with rasterio.open("file.tif") as rfh:
    data = rfh.read(1, window=window, masked=True)

print(data.shape)
```

Reply:
okay I simplified my approach to match this and the new tests pass (and I now have pre-commit set-up)